src/main/java/teammates/ui/controller/InstructorFeedbackResponseCommentsLoadAction.java

```diff
@@ -78,28 +78,36 @@ private void removeQuestionsAndResponsesIfNotAllowed(FeedbackSessionResultsBundl
         Iterator<FeedbackResponseAttributes> iter = bundle.responses.iterator();
         while (iter.hasNext()) {
             FeedbackResponseAttributes fdr = iter.next();
-            if (!(instructor != null &&
-                    instructor.isAllowedForPrivilege(fdr.giverSection, 
-                            fdr.feedbackSessionName, Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_SESSION_IN_SECTIONS)
-                    && instructor.isAllowedForPrivilege(fdr.recipientSection, 
-                            fdr.feedbackSessionName, Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_SESSION_IN_SECTIONS))) {
+            boolean instructorCanViewSessionInGiverSection = 
+                    instructor.isAllowedForPrivilege(fdr.giverSection, fdr.feedbackSessionName,
+                                       Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_SESSION_IN_SECTIONS);
+            boolean instructorCanViewSessionInRecipientSection =
+                    instructor.isAllowedForPrivilege(fdr.recipientSection, fdr.feedbackSessionName,
+                                       Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_SESSION_IN_SECTIONS);
+            
+            boolean instructorDoesNotHaveSessionViewingPrivileges = !(instructor != null
```

Comment:
it wont reach here, if instructor is null. It will throw NPE for instructor directly